### PR TITLE
idtoken: validate if token is expired

### PR DESCRIPTION
**Pull request by codyoss**
Updates: #484

### CODE REVIEW

File: idtoken/validate_test.go [hunk 1]
```diff
@@ -27,18 +28,27 @@ const (
...
+		nowFunc func() time.Time
 		wantErr bool
 	}{
-		{name: "works", keyID: keyID, n: pk.N, e: pk.E, wantErr: false},
-		{name: "no matching key", keyID: "5678", n: pk.N, e: pk.E, wantErr: true},
-		{name: "sig does not match", keyID: keyID, n: new(big.Int).SetBytes([]byte("42")), e: 42, wantErr: true},
+		{name: "works", keyID: keyID, n: pk.N, e: pk.E, nowFunc: beforeExp, wantErr: false},
+		{name: "no matching key", keyID: "5678", n: pk.N, e: pk.E, nowFunc: beforeExp, wantErr: true},
+		{name: "sig does not match", keyID: keyID, n: new(big.Int).SetBytes([]byte("42")), e: 42, nowFunc: beforeExp, wantErr: true},
+		{name: "token expired", keyID: keyID, n: pk.N, e: pk.E, nowFunc: afterExp, wantErr: true},
```
Comment (tbpg) [inline idtoken/validate_test.go:52]: Optional: might be easier to read these if they were on separate lines.

File: idtoken/validate_test.go [hunk 2]
```diff
@@ -64,7 +74,7 @@ func TestValidateRS256(t *testing.T) {
 					}
 				}),
 			}
-
+			now = tt.nowFunc
```
Comment (tbpg) [inline idtoken/validate_test.go:78]: Reset after the test? Don't want to accidentally mess up another test.

File: idtoken/validate_test.go [hunk 3]
```diff
@@ -27,18 +28,27 @@ const (
 	testAudience = "test-audience"
 )
 
+var (
+	// one second before the token expires
+	beforeExp = func() time.Time { return time.Unix(233431199, 0) }
```
Comment (broady) [inline idtoken/validate_test.go:33]: maybe

```
var (
	expiry    = ...
	beforeExp = func() time.Time { return exp + time.Second }
	afterExp  = func() time.Time { return exp + time.Second }
)
```

Comment (codyoss) [reply idtoken/validate_test.go:34]: Done

File: idtoken/validate_test.go [hunk 4]
```diff
@@ -27,18 +28,28 @@ const (
 	testAudience = "test-audience"
 )
 
+var (
+	expiry int64 = 233431200
+	// one second before the token expires
+	beforeExp = func() time.Time { return time.Unix(233431199, 0) }
+	// one second after the token expires
+	afterExp = func() time.Time { return time.Unix(233431201, 0) }
```
Comment (tbpg) [inline idtoken/validate_test.go:36]: Make this a const, then reference it in `beforeExp` and `afterExp`?

File: idtoken/validate_test.go (hunk 1 above)
Comment (codyoss) [reply idtoken/validate_test.go]: Good point. Done.

File: idtoken/validate_test.go (hunk 4 above)
Comment (codyoss) [reply idtoken/validate_test.go]: I like it, done.

File: idtoken/validate_test.go (hunk 2 above)
Comment (codyoss) [reply idtoken/validate_test.go:106]: Yes, done.

**Pull request merged**